test/testOnlyCPHL.m

```diff
@@ -0,0 +1,248 @@
+        xr420_reader = {@XRParse};
+
+        xr620_v000_param = getfield(chla_inputs(), 'xr620_v000_param')
+        xr620_reader = {@XRParse};
+
+        %sbs
+        sbe19p_v000_param = getfield(chla_inputs(), 'sbe19p_v000_param');
+        sbe19p_reader = {@SBE19Parse};
+
+        sbe25p_v000_param = getfield(chla_inputs(), 'sbe25p_v000_param');
+        sbe25p_reader = {@SBE19Parse};
+
+        sbe9p_v000_param = getfield(chla_inputs(), 'sbe9p_v000_param');
+        sbe9p_reader = {@SBE19Parse};
+
+        wqmraw_v000_param = getfield(chla_inputs(), 'wqmraw_v000_param');
+        wqmraw_reader = {@readWQMraw};
+
+        wqmdat_v000_param = getfield(chla_inputs(), 'wqmdat_v000_param');
+        wqmdat_reader = {@readWQMdat};
+    end
+
+    methods (Test)
+
+        function test_ecot(testCase, ecot_reader, ecot_v000_param, mode)
+            data = ecot_reader({ecot_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_jfe(testCase, jfe_reader, jfe_v000_param, mode)
+            data = jfe_reader({jfe_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_solo(testCase, solo_reader, solo_v000_param, mode)
+            data = solo_reader({solo_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        % TODO: investigate further files available not working
+        % function test_tr1060(testCase, tr1060_reader, tr1060_v000_param, mode)
+        %     data = tr1060_reader({tr1060_v000_param}, mode);
+        %     assert(check_cphl(data));
+        % end
+
+        function test_twr2050(testCase, twr2050_reader, twr2050_v000_param, mode)
+            data = twr2050_reader({twr2050_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_xr420(testCase, xr420_reader, xr420_v000_param, mode)
+            data = xr420_reader({xr420_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_xr620(testCase, xr620_reader, xr620_v000_param, mode)
+            data = xr620_reader({xr620_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_sbe19p(testCase, sbe19p_reader, sbe19p_v000_param, mode)
+            data = sbe19p_reader({sbe19p_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_sbe25p(testCase, sbe25p_reader, sbe25p_v000_param, mode)
+            data = sbe25p_reader({sbe25p_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_sbe9p(testCase, sbe9p_reader, sbe9p_v000_param, mode)
+            data = sbe9p_reader({sbe9p_v000_param}, mode);
+            assert(check_cphl(data));
+        end
+
+        function test_wqmraw(testCase, wqmraw_reader, wqmraw_v000_param, mode)
+            data = wqmraw_reader(wqmraw_v000_param, mode);
+            assert(check_cphl(data));
```

Comment:
This test should have picked up the CHLP typo?!

Reply:
If you run the test at the commit it was introduced, it should pass.